modules/behaviors/MapInteractionBehavior.js

```diff
@@ -362,4 +432,55 @@ export class MapInteractionBehavior extends AbstractBehavior {
     }
   }
 
+  /**
+   * _updatePinchState
+   * Updates the pinch state by recalculating the scale change and applying damping if necessary.
+   * This method is called during touch movements to dynamically adjust the map's scale.
+   */
+  _updatePinchState() {
+    const touchPoints = Object.values(this.activeTouches);
+    if (touchPoints.length === 2) {
+        const [first, second] = touchPoints;
+        const currentDistance = Math.hypot(first.x - second.x, first.y - second.y);
+
+        if (this._initialPinchDistance !== null) {
+            const scaleChange = currentDistance / this._initialPinchDistance;
+            const currentZoom = this.context.viewport.transform.zoom;
+
+            // Only apply changes if the scale change is significant
+            if (Math.abs(scaleChange - 1) > 0.01) {  // Adjust this threshold as needed
+                const baseDamping = 0.65;
+                const dampingFactor = currentZoom > 16 ? 0.25 : (currentZoom / 16) * baseDamping + 0.25;
+                const adjustedScaleChange = 1 + (scaleChange - 1) * dampingFactor;
+
+                this._applyPinchZoom(adjustedScaleChange);
```

Comment:
yeah, thinking more about this, I think it's probably safe to just get rid of `this._applyPinchZoom` and at this point just call `map.zoom(newZoom)`. Transform changes are already throttled by the renderer.